Dockerfile

```diff
@@ -0,0 +1,23 @@
+# Use a Rust base image
+FROM rust:latest
+
+# Install necessary tools (curl and tar)
+RUN apt-get update && apt-get install -y curl tar
+
+# Install mdBook
+RUN cargo install mdbook
+
+# Install mdbook-admonish
+RUN cargo install mdbook-admonish
```

Comment:
It is not needed to build the image on top of the rust image, nor compile the mdbook from scratch.

It would be better to use GitHub CLI (like [here](https://github.com/NilPointer-Software/bdfd-wiki/blob/dev/.github/workflows/book_deploy.yml#L20-L28)) or my [github-release.sh](https://gist.github.com/MineBartekSA/5234f09c1048f9ba007d0a26a8fd654e) script to download a pre-built binary.

This isn't a deal breaker (as in can stay as is), but I'm not too fond of docker images that build forever

Reply:
GitHub cli can only be used within GitHub as far as I'm aware, which is why it's not used here.